### PR TITLE
Add config to set the BigQuery Job timeout

### PR DESCRIPTION
**Pull request by isha97**

### CODE REVIEW

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java [hunk 1]
```diff
@@ -122,15 +122,16 @@ public static synchronized void runCleanupJobs() {
...
           job.waitFor(
               RetryOption.initialRetryDelay(Duration.ofSeconds(1)),
-              RetryOption.totalTimeout(Duration.ofMinutes(3)));
-      if (completedJob == null && completedJob.getStatus().getError() != null) {
+              RetryOption.totalTimeout(Duration.ofMinutes(jobTimeoutInMinutes)));
+      if (completedJob == null || completedJob.getStatus().getError() != null) {
         throw new UncheckedIOException(
-            new IOException(completedJob.getStatus().getError().toString()));
+            new IOException(
+                completedJob != null ? completedJob.getStatus().getError().toString() : null));
```
Comment (vishalkarve15) [inline bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:134]: When do we get the job as null? 
Is it on timeout? If it is, we should add the appropriate message instead.

File: spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java [hunk 2]
```diff
@@ -152,6 +152,8 @@ public static WriteMethod from(@Nullable String writeMethod) {
       "spark.sql.sources.partitionOverwriteMode";
 
   public PartitionOverwriteMode partitionOverwriteModeValue = PartitionOverwriteMode.STATIC;
+  public static final String BIGQUERY_JOB_TIMEOUT_IN_MINUTES = "bigQueryJobTimeoutInMinutes";
```
Comment (vishalkarve15) [inline spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/SparkBigQueryConfig.java:155]: Please add this property in the README template.

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java [hunk 3]
```diff
@@ -122,15 +122,16 @@ public static synchronized void runCleanupJobs() {
    *
    * @param job The {@code Job} to keep track of.
    */
-  public JobInfo waitForJob(Job job) {
+  public JobInfo waitForJob(Job job, long jobTimeoutInMinutes) {
```
Comment (vishalkarve15) [inline bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:125]: Should we instead make it a member of BigQueryClient? (Like `materializationDataset`)

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java (hunk 1 above)
Comment (isha97) [reply bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java:137]: `waitFor()` is a bigQuery API to get the job information which can return null.
We are throwing NPE here if the output is null.

File: README-template.md [hunk 4]
```diff
@@ -885,6 +885,15 @@ word-break:break-word
...
      <td>Read</td>
    </tr>
+  <tr>
+     <td><code>bigQueryJobTimeoutInMinutes</code>
+     </td>
+     <td>Config to set the BigQuery job timeout in minutes.
+          Default value is <code>360</code> minutes.
+         <br/> (Optional)
+     </td>
+     <td>Read</td>
```
Comment (vishalkarve15) [inline README-template.md:895]: This is for Read/Write both.

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**